src/layouts/_DevSiteSecondaryNavigationLayout.jsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { PageContainer } from '@cerner/terra-application/lib/page';
+import {
+  SecondaryNavigationLayout,
+  SecondaryNavigationGroup,
+  NavigationItem,
+  NavigationItemContext,
+} from '@cerner/terra-application/lib/layouts';
+import { useLocation, useHistory } from 'react-router-dom';
+
+import DevSitePage from '../pages/_DevSitePage';
+import NotFoundPage from '../pages/_NotFoundPage';
+import { contentImportsShape, navigationConfigShape } from '../site/siteConfigShapes';
+
+const propTypes = {
+  /**
+   * The unique id for secondary navigation layout.
+   */
+  id: PropTypes.string.isRequired,
+  /**
+   * The map linking the route to the content component to load.
+   */
+  contentImports: contentImportsShape.isRequired,
+  /**
+   * The navigation configuration describing the secondary navigation
+   */
+  config: navigationConfigShape,
+};
+
+const DevSiteSecondaryNavigationLayout = ({ id, config, contentImports }) => {
+  const location = useLocation();
+  const history = useHistory();
+  const { isActive } = React.useContext(NavigationItemContext);
+
+  if (!isActive) {
+    return null;
+  }
+
+  const retrieveNavItems = (navItems) => (
+    navItems.map((navItem) => {
+      if (navItem.children) {
+        return (
+          <SecondaryNavigationGroup
+            key={navItem.label}
+            label={navItem.label}
+          >
+            {retrieveNavItems(navItem.children)}
+          </SecondaryNavigationGroup>
+        );
+      }
+      return (
+        <NavigationItem
+          key={navItem.path}
+          navigationKey={navItem.path}
+          label={navItem.label}
+          renderPage={() => (
+            <DevSitePage pageContentConfig={navItem} contentImports={contentImports} />
+          )}
+        />
+      );
+    })
+  );
+
+  return (
+    <SecondaryNavigationLayout
+      id={id}
+      activeNavigationKey={location.pathname}
```

Comment:
I added a `label` prop to the SecondaryNavigationLayout that populates the header text within the side-nav panel. Previously it was defaulting to "Side Nav", which is lame. This label would likely want to match the primary navigation item label that it's associated to.

Reply:
I'll apply this in a future update since there are upcoming changes to terra-app v2 styles that will have to be accounted for beyond the current shaw this PR is locked into.